benchmarks/threadsafe_let_block.rb

```diff
@@ -0,0 +1,304 @@
+require 'rspec/core'
+
+# switches between these implementations - https://github.com/rspec/rspec-core/pull/1858/files
+# benchmark requested in this PR         - https://github.com/rspec/rspec-core/pull/1858
+#
+# I ran these by adding "benchmark-ips" to ../Gemfile
+# then exported BUNDLE_GEMFILE to point t it
+# then ran `bundle exec rspec threadsafe_let_block.rb`
+class LetBlockImplementationHelper
+
+  MemoizedHelpers = ::RSpec::Core::MemoizedHelpers
+  Memoized        = MemoizedHelpers::Memoized
+
+  def title(title)
+    hr    = "#" * (title.length + 6)
+    blank = "#  #{' ' * title.length}  #"
+    [hr, blank, "#  #{title}  #", blank, hr]
+  end
+
+  def use_non_threadsafe
+    return if @state == :old
+    @state = :old
+    MemoizedHelpers.module_eval do
+      def subject
+        __memoized.fetch(:subject) do
+          __memoized[:subject] = begin
+            described = described_class || self.class.metadata.fetch(:description_args).first
+            Class === described ? described.new : described
+          end
+        end
+      end
```

Comment:
I also noticed that the benchmarks create example groups and examples, which is additional overhead outside what we are trying to benchmark here.  I'm going to take a stab at revising this benchmark to deal with that.  I'll push a branch and let you know when I have something.